tests/cli/test_start_agent.py

```diff
@@ -58,6 +59,8 @@ async def agent_process():
 
 
 class TestAgentSignalForwarding:
+    # run these tests sequentially in the same xdist worker to fix flakiness
+    @pytest.mark.xdist_group(name="test_start_agent")
```

Comment:
@ddelange interesting — I thought we were doing xdist grouping by module already?

Reply:
oh interesting, in that case this should have no effect because the idea here was to not start multiple agent subprocesses concurrently. will try removing again

Reply:
We use

> --dist=loadscope: tests will be grouped by module for test functions and by class for test methods, then each group will be sent to an available worker, guaranteeing that all tests in a group run in the same process. This can be useful if you have expensive module-level or class-level fixtures. 

in CI